.github/workflows/docker-image-circleci.yml

```diff
@@ -1,4 +1,4 @@
-name: CircleCI Docker Container
+name: Build and Push Docker Image
```

Comment:
note that the name here is what shows up in the github ui so if multiple workflows have the same name they won't be distinguishable in the ui (also if the names are too long it just truncates them)